packages/android_intent_plus/example/lib/main.dart

```diff
@@ -109,6 +115,141 @@ class MyHomePage extends StatelessWidget {
     );
     intent.sendBroadcast();
   }
+
+  void _sendBroadCastExtra() {
+    final intent = AndroidIntent(
+      action: 'com.symbol.datawedge.api.ACTION',
+      extras: [
+        Bundle(
+          value: [
+            PutBundle(
+              key: 'com.symbol.datawedge.api.SET_CONFIG',
+              value: [
+                PutString(
+                  key: 'PROFILE_NAME',
+                  value: 'com.dalosy.scanner_by_intents',
+                ),
+                PutString(
+                  key: 'PROFILE_ENABLED',
+                  value: 'true',
+                ),
+                PutString(
+                  key: 'CONFIG_MODE',
+                  value: 'CREATE_IF_NOT_EXIST',
+                ),
+                PutParcelableArray(
+                  key: 'APP_LIST',
+                  value: [
+                    appList(),
+                  ],
+                ),
+                PutParcelableArrayList(
+                  key: 'PLUGIN_CONFIG',
+                  value: [
+                    barcodePluginConfig(),
+                    intentPluginConfig(),
+                    keystrokePluginConfig(),
```

Comment:
Is there a benefit of using methods over objects (also `appList`)?

Reply:
Hi, 
The sample code is purely based on the use case described here:
https://techdocs.zebra.com/datawedge/11-4/guide/api/setconfig/#nestedbundles.

The nesting goes a whole lot deeper than is visualized in my example. The functions represent a seperate part of the data structure and are only there to improve the readability and seperate concerns.

Reply:
Makes sense.

I meant, instead of this:

```
  Bundle appList() => Bundle(
        value: [
          PutString(
            key: 'PACKAGE_NAME',
            value: 'com.dalosy.scanner_by_intents',
          ),
          PutStringArray(
            key: 'ACTIVITY_LIST',
            value: ['*'],
          ),
        ],
      );
```

like this:
```
  final appList = Bundle(
    value: [
      PutString(
        key: 'PACKAGE_NAME',
        value: 'com.dalosy.scanner_by_intents',
      ),
      PutStringArray(
        key: 'ACTIVITY_LIST',
        value: ['*'],
      ),
    ],
  );
```

Reply:
Hi,
Ah, right. No, I don't think there is any benefit in using one or the other. (Is there?)

Reply:
The function creates a new object every time it's called, an object would be reused.

Since we only need to create the object once and don't need to change it (no parameters), declaring a variable could be simpler and more efficient.

Another suggestion: They could be private.

Reply:
Of course, you are right. This is part of the sample code and I did not think about optimizing. It just came out of my fingers that way.